packages/frontend/src/scripts/configureFilters.ts

```diff
@@ -49,11 +63,80 @@ export function configureFilters() {
         !(archivedCheckbox?.checked && combinedCheckbox.checked),
       )
     })
+    renderNumbers()
   })
 
   upcomingCheckbox?.addEventListener('change', () => {
+    localStorage.setItem(
+      'l2beat-filter-upcoming',
+      upcomingCheckbox.checked.toString(),
+    )
+
     upcoming.forEach((x) => {
       x.classList.toggle('hidden', !upcomingCheckbox.checked)
     })
+    renderNumbers()
+  })
+}
+
+function setupArchivedCheckbox(archived: HTMLElement[]) {
+  const saved = localStorage.getItem('l2beat-filter-archived')
+
+  const archivedCheckbox =
+    document.querySelector<HTMLInputElement>('#archived-projects')
+
+  if (archivedCheckbox) {
+    archivedCheckbox.checked = saved === 'true' ? true : false
+
+    archived.forEach((x) => {
+      x.classList.toggle('hidden', !archivedCheckbox.checked)
+    })
+  }
+
+  return archivedCheckbox
+}
+
+function setupUpcomingCheckbox(upcoming: HTMLElement[]) {
+  const saved = localStorage.getItem('l2beat-filter-upcoming')
+
+  const upcomingCheckbox =
+    document.querySelector<HTMLInputElement>('#upcoming-rollups')
+
+  if (upcomingCheckbox) {
+    upcomingCheckbox.checked = saved === 'true' ? true : false
+
+    upcoming.forEach((x) => {
+      x.classList.toggle('hidden', !upcomingCheckbox.checked)
+    })
+  }
+
+  return upcomingCheckbox
+}
```

Comment:
This could have been one function with 3 parameters, but it's not super crucial (we'll be rewriting filters anyway soon)